app/services/builders/metric_chart/base.rb

```diff
@@ -70,6 +70,18 @@ def metric_ownable_type
       def metric_name
         raise NoMethodError
       end
+
+      def metric_interval
+        raise NoMethodError
+      end
+
+      def chart_date_format
+        raise NoMethodError
+      end
+
+      def grouping_period
+        raise NoMethodError
+      end
```

Comment:
I'm wondering which is the difference of not having this, if you call a method that is not implemented you are going to get a `NoMethodError` anyway, don't you?

Reply:
Yes, you're totally correct. What we discussed is the necessity of having some live documentation in our code. Defining these methods like this is surely the same as not doing so from an execution standpoint, but making it explicit like this makes the devs understand better which are the methods we should be implementing in the subclasses.

Reply:
I had some discussions about this in other projects. Why are we trying to implement some interfaces like other languages ​​(like Java)? I think it's an additional layer that we don't need, I know the meaning is documenting, but for ruby on rails ​​developers it is not necessary in my opinion

Reply:
Alright, given that none of us is sure about this approach and that the rest of the code is not written this way, I'm just gonna delete all these abstract methods.
But I'll do that in other PR.